index.js

```diff
@@ -12,12 +13,14 @@ const register = (server, pluginOptions) => {
   };
 
   server.ext('onRequest', (request, h) => {
-    if (request.path.endsWith('.html')) {
-      const query = request.query;
-      request.headers.accept = 'text/html';
-      request.setUrl(request.path.replace('.html', ''));
-      request.query = query;
+    const query = request.query;
```

Comment:
seems like you removed the path.endsWith check - which tells me there is no test for this case